### PR TITLE
Fix for "Cave Revamp" update

### PR DESCRIPTION
**Pull request by developer9998**
- Fix object locations for new update

### CODE REVIEW

**Comment by developer9998** [comment]
- Now uses ZoneManagement and ZoneData to get the selectors
- Code probably needs polish, but as of now all the selectors work fine and should work fine without CI

File: Utilla/GamemodeSelector.cs [hunk 1]
```diff
@@ -67,7 +68,7 @@ GameObject CreatePageButton(string text, Action onPressed)
 
 				if (!button.GetComponentInParent<Canvas>())
 				{
-					Canvas canvas = button.transform.parent.gameObject.AddComponent<Canvas>();
+					Canvas canvas = button.transform.parent.gameObject.GetOrAddComponent<Canvas>();
```
Comment (Graicc) [inline Utilla/GamemodeSelector.cs:71]: Doesn't `!button.GetComponentInParent<Canvas>()` guarantees that `GetOrAddComponent` will always be the same as `AddComponent`?

Comment (developer9998) [reply Utilla/GamemodeSelector.cs:71]: i thought of that too, but without the code i had there would be errors thrown for i'm pretty sure mountain clouds and beach showing that a canvas already exists

Comment (Graicc) [reply Utilla/GamemodeSelector.cs:71]: Huh weird, I got the same error though

**Pull request merged**